torch/csrc/jit/init.cpp

```diff
@@ -120,6 +120,9 @@ void initJITBindings(PyObject *module) {
       });
 
   py::class_<ExecutionPlanState>(m, "ExecutionPlanState")
+    .def("_graph_copy", [](ExecutionPlanState& s) {
+      return s.graph->copy();
+    })
```

Comment:
Can we make `assertExpectedGraph` *not* mutate the inputs? That's very counterintuitive, and shouldn't really happen. Let's just make it clone the graph when it's called (and expose `Graph::copy()` if it's not exposed yet).